fluent-dom/src/document_localization.js

```diff
@@ -0,0 +1,179 @@
+import Localization from './localization';
+
+export default class DocumentLocalization extends Localization {
+  constructor(doc, resIds, generateMessage) {
+    super(doc.location.href, resIds, generateMessage);
+    this.document = doc;
+    this.query = '[data-l10n-id]';
+    this.roots = new Set();
+    this.mutationObserver = new doc.defaultView.MutationObserver(
+      mutations => this.translateMutations(mutations)
+    );
+
+    this.observerConfig = {
+      attribute: true,
+      characterData: false,
+      childList: true,
+      subtree: true,
+      attributeFilter: ['data-l10n-id', 'data-l10n-args']
+    };
+  }
+
+  onLanguageChange() {
+    super.onLanguageChange();
```

Comment:
`Localization` doesn't seem to implement `onLanguageChange`. Also, it might be a good idea to make `DocumentLocalization` implement the `EventListener` interface via the `handleEvent` method.